homeassistant/components/isy994/button.py

```diff
@@ -36,47 +37,62 @@ async def async_setup_entry(
     async_add_entities(entities)
 
 
-class ISYNodeQueryButtonEntity(ButtonEntity):
-    """Representation of a device query button entity."""
+class ISYNodeButtonEntity(ButtonEntity):
+    """Additional methods for ISYNodeButton classes."""
 
     _attr_should_poll = False
-    _attr_entity_category = EntityCategory.CONFIG
     _attr_has_entity_name = True
 
     def __init__(self, node: Node | ISY, base_unique_id: str) -> None:
-        """Initialize a query ISY device button entity."""
+        """Initialize a ISY device button entity."""
         self._node = node
+        self._attr_entity_registry_enabled_default = getattr(node, "enabled", True)
+        self._attr_device_info = DeviceInfo(
+            identifiers={(ISY994_DOMAIN, base_unique_id)}
+        )
+
+    def check_disabled(self) -> bool:
```

Comment:
```suggestion
    def _raise_if_disabled(self) -> bool:
```

I'd probably name it something like the above